models/work-item-link-category-repository.go

```diff
@@ -0,0 +1,169 @@
+	id, err := satoriuuid.FromString(ID)
+	if err != nil {
+		// treat as not found: clients don't know it must be a UUID
+		return nil, NewNotFoundError("work item link category", ID)
+	}
+	log.Printf("loading work item link category %s", id.String())
+	res := WorkItemLinkCategory{}
+	db := r.db.Model(&res).Where("id=?", ID).First(&res)
+	if db.RecordNotFound() {
+		log.Printf("not found, res=%v", res)
+		return nil, NewNotFoundError("work item link category", id.String())
+	}
+
+	// Convert the created link category entry into a JSONAPI response
+	result := ConvertLinkCategoryFromModel(&res)
+	return &result, nil
+}
+
+// List returns all work item link categories
+// TODO: Handle pagination
+func (r *GormWorkItemLinkCategoryRepository) List(ctx context.Context) (*app.WorkItemLinkCategoryArray, error) {
+	var rows []WorkItemLinkCategory
+	db := r.db.Find(&rows)
+	if db.Error != nil {
+		return nil, db.Error
+	}
+	res := app.WorkItemLinkCategoryArray{}
+	res.Data = make([]*app.WorkItemLinkCategoryData, len(rows))
+	for index, value := range rows {
+		cat := ConvertLinkCategoryFromModel(&value)
+		res.Data[index] = cat.Data
+	}
+	// TODO: When adding pagination, this must not be len(rows) but
+	// the overall total number of elements from all pages.
+	res.Meta = &app.WorkItemLinkCategoryArrayMeta{
+		TotalCount: len(rows),
+	}
+	return &res, nil
+}
+
+// Delete deletes the work item link category with the given id
+// returns NotFoundError or InternalError
+func (r *GormWorkItemLinkCategoryRepository) Delete(ctx context.Context, ID string) error {
+	id, err := satoriuuid.FromString(ID)
+	if err != nil {
+		// treat as not found: clients don't know it must be a UUID
+		return NewNotFoundError("work item link category", ID)
+	}
+
+	var cat = WorkItemLinkCategory{
+		ID: id,
+	}
+
+	log.Printf("work item link category to delete %v\n", cat)
+
+	db := r.db.Delete(&cat)
+	if db.Error != nil {
+		return NewInternalError(db.Error.Error())
+	}
+
+	if db.RowsAffected == 0 {
+		return NewNotFoundError("work item link category", id.String())
+	}
+	return nil
+}
+
+// Save updates the given work item link category in storage. Version must be the same as the one int the stored version.
+// returns NotFoundError, VersionConflictError, ConversionError or InternalError
+func (r *GormWorkItemLinkCategoryRepository) Save(ctx context.Context, linkCat app.WorkItemLinkCategory) (*app.WorkItemLinkCategory, error) {
+	res := WorkItemLinkCategory{}
+	if linkCat.Data.ID == nil {
+		return nil, NewBadParameterError("data.id", linkCat.Data.ID)
+	}
+	id, err := satoriuuid.FromString(*linkCat.Data.ID)
+	if err != nil {
+		//log.Printf("Error when converting %s to UUID: %s", *linkCat.Data.ID, err.Error())
+		// treat as not found: clients don't know it must be a UUID
+		return nil, NewNotFoundError("work item link category", id.String())
+	}
+
+	if linkCat.Data.Type != EndpointWorkItemLinkCategories {
```

Comment:
I know we need to move things around. And in fact, the request handler does check for this using an [enum-validation](https://github.com/kwk/almighty-core/blob/81d19767dce0d6a4c6b0a58009d1d33f54937878/design/user_types.go#L205):
```go
	a.Attribute("type", d.String, func() {
		a.Enum("workitemlinkcategories")
	})
``` 
So theoretically, there's no need to check it here. But I'd like to postpone this optimization for now until I have finished the overall task. Okay? Moving everything with `app.*` out of the models folder should be done in a separate step.